File: app/src/main/java/com/lumiyaviewer/lumiya/render/ModernTextureManagerTest.java
package com.lumiyaviewer.lumiya.render;

import android.content.Context;
import android.util.Log;
import java.io.ByteArrayInputStream;
import java.io.IOException;

/**
 * Test class for ModernTextureManager functionality
 */
public class ModernTextureManagerTest {
    private static final String TAG = "ModernTextureManagerTest";
    
    public static void runBasicTests(Context context) {
        Log.i(TAG, "Starting ModernTextureManager tests...");
        
        try {
            // Test initialization
            testInitialization(context);
            
            // Test GPU capability detection
            testGPUCapabilities(context);
            
            // Test format utilities
            testFormatUtilities();
            
            Log.i(TAG, "All tests completed successfully");
            
        } catch (Exception e) {
            Log.e(TAG, "Test failed with exception", e);
        }
    }
    
    private static void testInitialization(Context context) {
        Log.i(TAG, "Testing initialization...");
        
        ModernTextureManager manager = new ModernTextureManager(context);
        if (manager != null) {
            Log.i(TAG, "✓ ModernTextureManager created successfully");
        } else {
            Log.e(TAG, "✗ Failed to create ModernTextureManager");
        }
    }
    
    private static void testGPUCapabilities(Context context) {
        Log.i(TAG, "Testing GPU capability detection...");
        
        ModernTextureManager manager = new ModernTextureManager(context);
        int optimalFormat = manager.getOptimalTextureFormat();
        String formatName = ModernTextureManager.getFormatName(optimalFormat);
        
        Log.i(TAG, "✓ Optimal texture format detected: " + formatName + " (" + optimalFormat + ")");
    }
    
    private static void testFormatUtilities() {
        Log.i(TAG, "Testing format utilities...");
        
        // Test format name mapping
        String astcName = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_ASTC_4x4_RGBA);
        String etc2Name = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_ETC2_RGBA);
        String bc7Name = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_BC7_RGBA);
        String rgbaName = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_RGBA32);
        
        Log.i(TAG, "✓ Format names: ASTC=" + astcName + ", ETC2=" + etc2Name + 
                   ", BC7=" + bc7Name + ", RGBA=" + rgbaName);
        
        // Test OpenGL format mapping
        int astcGL = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_ASTC_4x4_RGBA);
        int etc2GL = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_ETC2_RGBA);
        int bc7GL = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_BC7_RGBA);
        int rgbaGL = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_RGBA32);
        
        Log.i(TAG, "✓ OpenGL formats: ASTC=0x" + Integer.toHexString(astcGL) + 
                   ", ETC2=0x" + Integer.toHexString(etc2GL) + 
                   ", BC7=0x" + Integer.toHexString(bc7GL) + 
                   ", RGBA=0x" + Integer.toHexString(rgbaGL));
    }
    
    private static void testTextureDataClass() {
        Log.i(TAG, "Testing TextureData class...");
        
        byte[] testData = new byte[1024];
        ModernTextureManager.TextureData textureData = new ModernTextureManager.TextureData(
                256, 256, 1, ModernTextureManager.FORMAT_ASTC_4x4_RGBA, testData
        );
        
        Log.i(TAG, "✓ TextureData created: " + textureData.toString());
        Log.i(TAG, "✓ Is compressed: " + textureData.isCompressed());
        Log.i(TAG, "✓ Format name: " + textureData.getFormatName());
        Log.i(TAG, "✓ OpenGL format: 0x" + Integer.toHexString(textureData.getOpenGLFormat()));
    }
}

File: Linkpoint/src/main/kotlin/com/linkpoint/render/ModernTextureManagerTest.kt
package com.linkpoint.render

import android.content.Context
import android.util.Log
import java.io.ByteArrayInputStream
import java.io.IOException

/**
 * Test class for ModernTextureManager functionality
 */
class ModernTextureManagerTest {
    private const val TAG: String = "ModernTextureManagerTest"
    
    @JvmStatic
     fun runBasicTests(context: Context) {
        Log.i(TAG, "Starting ModernTextureManager tests...")
        
        try {
            // Test initialization
            testInitialization(context)
            
            // Test GPU capability detection
            testGPUCapabilities(context)
            
            // Test format utilities
            testFormatUtilities()
            
            // Test TextureData class
            testTextureDataClass()
            
            Log.i(TAG, "All tests completed successfully")
            
        } catch (Exception e) {
            Log.e(TAG, "Test failed with exception", e)
        }
    }
    
    @JvmStatic
 private fun testInitialization(context: Context) {
        Log.i(TAG, "Testing initialization...")
        
        try {
            val manager: ModernTextureManager = ModernTextureManager(context)
            if (manager != null && manager.isInitialized()) {
                Log.i(TAG, "✓ ModernTextureManager created and initialized successfully")
            } else {
                Log.e(TAG, "✗ ModernTextureManager not properly initialized")
            }
        } catch (Exception e) {
            Log.e(TAG, "✗ Failed to create ModernTextureManager", e)
        }
    }
    
    @JvmStatic
 private fun testGPUCapabilities(context: Context) {
        Log.i(TAG, "Testing GPU capability detection...")
        
        try {
            val manager: ModernTextureManager = ModernTextureManager(context)
            if (manager.isInitialized()) {
                val optimalFormat: Int = manager.getOptimalTextureFormat()
                val formatName: String = ModernTextureManager.getFormatName(optimalFormat)
                
                Log.i(TAG, "✓ Optimal texture format detected: " + formatName + " (" + optimalFormat + ")")
            } else {
                Log.e(TAG, "✗ Manager not initialized, cannot test capabilities")
            }
        } catch (Exception e) {
            Log.e(TAG, "✗ Failed to test GPU capabilities", e)
        }
    }
    
    @JvmStatic
 private fun testFormatUtilities() {
        Log.i(TAG, "Testing format utilities...")
        
        // Test format name mapping
        val astcName: String = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_ASTC_4x4_RGBA)
        val etc2Name: String = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_ETC2_RGBA)
        val bc7Name: String = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_BC7_RGBA)
        val rgbaName: String = ModernTextureManager.getFormatName(ModernTextureManager.FORMAT_RGBA32)
        
        Log.i(TAG, "✓ Format names: ASTC=" + astcName + ", ETC2=" + etc2Name + 
                   ", BC7=" + bc7Name + ", RGBA=" + rgbaName)
        
        // Test OpenGL format mapping
        val astcGL: Int = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_ASTC_4x4_RGBA)
        val etc2GL: Int = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_ETC2_RGBA)
        val bc7GL: Int = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_BC7_RGBA)
        val rgbaGL: Int = ModernTextureManager.getOpenGLFormat(ModernTextureManager.FORMAT_RGBA32)
        
        Log.i(TAG, "✓ OpenGL formats: ASTC=0x" + Integer.toHexString(astcGL) + 
                   ", ETC2=0x" + Integer.toHexString(etc2GL) + 
                   ", BC7=0x" + Integer.toHexString(bc7GL) + 
                   ", RGBA=0x" + Integer.toHexString(rgbaGL))
    }
    
    @JvmStatic
 private fun testTextureDataClass() {
        Log.i(TAG, "Testing TextureData class...")
        
        val testData: ByteArray = Byte[1024]
        ModernTextureManager.TextureData textureData = ModernTextureManager.TextureData(
                256, 256, 1, ModernTextureManager.FORMAT_ASTC_4x4_RGBA, testData
        )
        
        Log.i(TAG, "✓ TextureData created: " + textureData.toString())
        Log.i(TAG, "✓ Is compressed: " + textureData.isCompressed())
        Log.i(TAG, "✓ Format name: " + textureData.getFormatName())
        Log.i(TAG, "✓ OpenGL format: 0x" + Integer.toHexString(textureData.getOpenGLFormat()))
    }
}